### PR TITLE
feat(interactivity): Add support for components in pages

### LINKED ISSUE #1966: Ability to add our own additional components to a SendPaginatedResponseAsync
### Description

Use case:

I have a large list of things that require a paged response to display it appropriately, and D#+'s  SendPaginatedResponseAsync is beautiful for this purpose and saves me sooooo much time.

However, should a user need to pick an item from said paged response to provide further interactivity, this is not currently possible.

Is there a way that we could provide our own components to SendPaginatedResponseAsync, await for their interaction, and then for us to respond to our own components.

### Specify the libraries you want this feature request to affect

DSharpPlus.Interactivity

### Other considerations

_No response_

### PR DESCRIPTION
**Pull request by VelvetToroyashi**
Closes #1966

### CODE REVIEW

File: DSharpPlus.Interactivity/InteractivityExtension.cs [hunk 1]
```diff
@@ -769,20 +771,80 @@ public async Task<ReadOnlyCollection<Reaction>> CollectReactionsAsync(DiscordMes
...
+        IEnumerable<DiscordMessageBuilder> pageBuilders,
+        PaginationButtons buttons,
+        PaginationBehaviour? behaviour = default,
+        ButtonPaginationBehavior? deletion = default,
+        CancellationToken token = default
+    )
+    {
+        PaginationBehaviour bhv = behaviour     ?? this.Config.PaginationBehaviour;
+        ButtonPaginationBehavior del = deletion ?? this.Config.ButtonBehavior;
+        PaginationButtons bts = buttons         ?? this.Config.PaginationButtons;
```
Comment (akiraveliara) [inline DSharpPlus.Interactivity/InteractivityExtension.cs:810]: naming, whitespace style

File: DSharpPlus.Interactivity/InteractivityExtension.cs [hunk 2]
```diff
@@ -769,20 +771,80 @@ public async Task<ReadOnlyCollection<Reaction>> CollectReactionsAsync(DiscordMes
...
+
+        bts = new(bts);
```
Comment (akiraveliara) [inline DSharpPlus.Interactivity/InteractivityExtension.cs:812]: why?

File: DSharpPlus.Interactivity/InteractivityExtension.cs (hunk 1 above)
Comment (VelvetToroyashi) [reply DSharpPlus.Interactivity/InteractivityExtension.cs:810]: c&p

File: DSharpPlus.Interactivity/EventHandling/Requests/PaginationEmojis.cs [hunk 3]
```diff
@@ -25,9 +28,65 @@ public class Page
     public string Content { get; set; }
     public DiscordEmbed Embed { get; set; }
 
-    public Page(string content = "", DiscordEmbedBuilder embed = null)
+    public IReadOnlyList<DiscordActionRowComponent> Components { get; }
+
+    public Page(IDiscordMessageBuilder builder)
+    : this(builder.Content, builder.Embeds.FirstOrDefault(), builder.Components) { }
```
Comment (Plerx2493) [inline DSharpPlus.Interactivity/EventHandling/Requests/PaginationEmojis.cs:34]: This should also be yeeted

**Review by Plerx2493 — CHANGES_REQUESTED** [review]

**Review by Plerx2493 — APPROVED** [review]

**Pull request merged**